src/cmdstan/diagnose.cpp

```diff
@@ -0,0 +1,189 @@
+#include <algorithm>
+#include <iostream>
+#include <iomanip>
+#include <ios>
+#include <stan/mcmc/chains.hpp>
+#include <cmdstan/stansummary_helper.hpp>
+#include <fstream>
+
+void diagnose_usage() {
+  std::cout << "USAGE:  diagnose <filename 1> [<filename 2> ... <filename N>]"
+            << std::endl
+            << std::endl;
+}
+
+/**
+ * The Stan print function.
+ *
+ * @param argc Number of arguments
+ * @param argv Arguments
+ *
+ * @return 0 for success,
+ *         non-zero otherwise
+ */
+int main(int argc, const char* argv[]) {
+
+  if (argc == 1) {
+    diagnose_usage();
+    return 0;
+  }
+
+  // Parse any arguments specifying filenames
+  std::ifstream ifstream;
+  std::vector<std::string> filenames;
+
+  for (int i = 1; i < argc; i++) {
+    ifstream.open(argv[i]);
+
+    if (ifstream.good()) {
+      filenames.push_back(argv[i]);
+      ifstream.close();
+    } else {
+      std::cout << "File " << argv[i] << " not found" << std::endl;
```

Comment:
stan_csv_reader will throw if the files are ill-formatted.  It's the same error handling as in stansummary.